src/api/services/topology.ts

```diff
@@ -130,6 +130,26 @@ export const getTopologyComponentsWithLogs = () => {
   );
 };
 
+type AgentItem = {
+  id: string;
+  name: string;
+  description: string;
+};
+
+export const getAgentByID = async (id: string) => {
+  const res = await IncidentCommander.get<AgentItem[] | null>(
```

Comment:
Is this going to be cached ?

Reply:
Yeah, we are using react query, so it should be cached and only fetched once, even if you send multiple requests at once.